pennylane/qchem/integrals.py

```diff
@@ -304,13 +332,16 @@ def _overlap_integral(*args):
         Returns:
             array[float]: the overlap integral between two contracted Gaussian orbitals
         """
-
         args_a = [arg[0] for arg in args]
         args_b = [arg[1] for arg in args]
+
+        # Generate params returns params in alpha, coeff, coord order
         alpha, ca, ra = _generate_params(basis_a.params, args_a)
         beta, cb, rb = _generate_params(basis_b.params, args_b)
 
-        if basis_a.params[1].requires_grad or normalize:
+        if _check_requires_grad(
+            basis_param=basis_a.params[1], normalize=normalize, args=args, index=1
+        ):
```

Comment:
do we need to use all they keyword names? (maybe just sending arguments to make this cleaner)

Reply:
Could just send the arguments yea

Reply:
I would prefer that, would look cleaner.